README.md

```diff
@@ -1,67 +1,94 @@
+- System-versioned MariaDB database,
+- Python web API to manage permissions, and store frequently used queries,
+- An installable python library that wraps the Python web API (using OpenAPI generator)
 
-### Running the web server
+Every resource in sample-metadata belongs to a project. All resources are access
+controlled through membership of the google groups:
+`$dataset-sample-metadata-main-{read,write}`. Note that members of google-groups
+are cached in a secret as group-membership identity checks are slow.
 
-The default behaviour for the web server is to:
 
-- Use a GCP secret for the DB credentials
+## Structure
 
-You can configure this with:
 
-```shell
-# use credentials defined in db/python/connect.py:dev_config
-export SM_ENVIRONMENT=LOCAL
-# use specific mysql settings
-export SM_DEV_DB_PROJECT=sm_dev
-export SM_DEV_DB_USER=root
-export SM_DEV_DB_PORT=3307
-export SM_DEV_DB_HOST=127.0.0.1
-export SM_DEV_DB_PASSWORD=root
-```
+## Sample IDs
+
+In an effort to reduce our dependency on potentially mutable external sample IDs with inconsistent format,
+the sample-metadata server generates an internal sample id for every sample. Internally they're an
+incrementing integer, but they're transformed externally to have a prefix, and checksum - this allows durability
+when transcribing sample IDs to reduce mistypes, and allow to quickly check whether a sample ID is valid.
+
+> NB: The prefix and checksums are modified per environment (production, development, local) to avoid duplicates from these environments.
+
+For example, let's consider the production environment which uses the prefix of `CPG` and a checksum offset of 0:
+
+> A sample is given the internal ID `12345`, we calculate the Luhn checksum to be `5` (with no offset applied).
+> We can then concatenate the results, for the final sample ID to be `CPG123455`.
+
+
+## Local develompent of SM
 
-### Using the Python API
+The recommended way to develop the sample-metadata system is to run a local copy of SM.
+> There have been some reported issues of running a local SM environment on an M1 mac.
 
-The default behaviour for the installable Python API is to:
+You can run MariaDB with a locally installed docker, or from within a docker container.
+You can configure the MariaDB connection with environment variables.
 
-- Use the production server
-- With a service account
+### Creating the environment
 
-You can configure these separately with:
+You should use `conda` to manage the SM environment, as some of the required
+packages are only available on Conda.
+
+The conda env defines the development dependencies, and the pure API deps
+are defined in `requirements.txt`.
 
 ```shell
-# use localhost:8000
-export SM_ENVIRONMENT=LOCAL
- # use a local account to generate identity-token for auth
-export GOOGLE_APPLICATION_CREDENTIALS=/path/to/sa.json
+conda env create -f environment-dev.yml
+conda activate sample-metadata
```

Comment:
To match the name specified in `environment-dev.yml`
```suggestion
conda activate sample-metadata-dev
```

Reply:
This should've been changed to `sample-metadata` in environment-dev: https://github.com/populationgenomics/sample-metadata/blob/6bef596a9870f1caab39f8d11d7f2c80907720ff/environment-dev.yml

Reply:
oh, sorry I was in the wrong branch!